judge/admin/submission.py

```diff
@@ -124,6 +124,9 @@ def has_change_permission(self, request, obj=None):
             return True
         return obj.problem.is_editor(request.user.profile)
 
+    def lookup_allowed(self, key, value):
+        return super(SubmissionAdmin, self).lookup_allowed(key, value) or key in ['problem__code']
```

Comment:
Using a tuple is more idiomatic in this case.

Reply:
Tuples are also faster for this kind of thing, but that's not a significant concern. It is more idiomatic and that's what should matter.